File: examples/node/consulta.mjs
const apiBase = process.env.PATENTE_API_BASE_URL ?? "https://api.patente.ar/v1/consultas";
const apiKey = process.env.PATENTE_API_KEY;
const idempotencyKey = process.env.PATENTE_IDEMPOTENCY_KEY ?? "vtv-demo-001";
const xToken = process.env.PATENTE_X_TOKEN ?? "orden-externa-001";

if (!apiKey) {
  throw new Error("Set PATENTE_API_KEY before running this example.");
}

const payload = {
  "patente": [
    "AB123CD"
  ],
  "jurisdiccion": [
    "pba",
    "caba"
  ]
};

const response = await fetch(apiBase, {
  method: "POST",
  headers: {
    "authorization": `Bearer ${apiKey}`,
    "content-type": "application/json",
    "idempotency-key": idempotencyKey,
    "x-token": xToken,
  },
  body: JSON.stringify(payload),
});

const body = await response.json().catch(() => null);
console.log(JSON.stringify({ status: response.status, body }, null, 2));

File: examples/python/consulta.py
#!/usr/bin/env python3
import json
import os
import urllib.request

api_base = os.environ.get("PATENTE_API_BASE_URL", "https://api.patente.ar/v1/consultas")
api_key = os.environ.get("PATENTE_API_KEY")
idempotency_key = os.environ.get("PATENTE_IDEMPOTENCY_KEY", "vtv-demo-001")
x_token = os.environ.get("PATENTE_X_TOKEN", "orden-externa-001")

if not api_key:
    raise SystemExit("Set PATENTE_API_KEY before running this example.")

payload = {
  "patente": [
    "AB123CD"
  ],
  "jurisdiccion": [
    "pba",
    "caba"
  ]
}

request = urllib.request.Request(
    api_base,
    data=json.dumps(payload).encode("utf-8"),
    method="POST",
    headers={
        "Authorization": f"Bearer {api_key}",
        "Content-Type": "application/json",
        "Idempotency-Key": idempotency_key,
        "x-token": x_token,
    },
)

try:
    with urllib.request.urlopen(request, timeout=30) as response:
        print(response.status)
        print(response.read().decode("utf-8"))
except urllib.error.HTTPError as error:
    print(error.code)
    print(error.read().decode("utf-8"))
